figo/figo.py

```diff
@@ -293,690 +195,764 @@ class FigoConnection(FigoObject):
+                dict with lists of supported banks and payment services
         """
-        return self._query_api_object(Account, "/rest/accounts/%s" % account.account_id,
-                                      account.dump(), "PUT")
+        options = urlencode(filter_none({"country": country_code}))
 
-    def remove_account(self, account_or_account_id):
-        """Remove an account.
+        catalog = self._request_with_exception(f"/rest/catalog?{options}")
 
-        Args:
-            account_or_account_id: account to be removed or its ID
-        """
-        path = "/rest/accounts/{0}".format(getAccountId(account_or_account_id))
-        self._request_with_exception(path, method="DELETE")
+        for k, v in catalog.items():
+            catalog[k] = self._process_catalog_list(v)
+        return catalog
+
+    # TODO: Missing unit test but used in ownly-backend
+    # TODO: Maybe we should use directly FigoBank instead of this method.
+    # TODO: Looks like this is more FigoConnection method
+    def get_login_settings(self, country_code, item_id):
+        """Return the login settings of a bank.
 
-    def add_sync(self, access_id, disable_notifications, redirect_uri, state, credentials, save_secrets):
-        """
         Args:
-          access_id (str): figo ID of the provider access, Required
-          disable_notifications (bool): This flag indicates whether notifications should be sent to your
-            application, Optional, default: False
-          redirect_uri (str): The URI to which the end user is redirected in OAuth cases, Optional
-          state (str): Arbitrary string to maintain state between this request and the callback
-          credentials (obj): Credentials used for authentication with the financial service provider.
-          save_secrets (bool): Indicates whether the confidential parts of the credentials should be saved, default: False
+            country_code (str): country code of the requested bank
+            item_id (str): bank code or fake bank code of the requested bank
 
         Returns:
-          Object: synchronization operation.
+            LoginSettings: Object that contains information which are needed
+                for logging in to the bank
         """
-        data = filterNone({ "disable_notifications": disable_notifications, "redirect_uri": redirect_uri,
-                 "state": state, "credentials": credentials, "save_secrets": save_secrets})
+        query_params = urlencode({"country": country_code, "q": item_id})
 
-        return self._query_api_object(Sync, "/rest/accesses/{0}/syncs".format(access_id), data=data, method='POST')
+        # now the catalog returns matches for all possible banks
+        response = self._query_api_object(
+            LoginSettings,
+            f"/catalog/banks?{query_params}",
+            collection_name="collection",
+        )
+        if response and len(response) > 0:
```

Comment:
Second condition appears to be redundant.
```suggestion
        if response:
```